### ISSUE TITLE
Shapefile export attribute data limited to hardcoded 255 character length

### ISSUE THREAD

**Issue by samuelscandrett** [archesproject/arches#12679]
(identified in 7.6 and 8.x)

The Arches search export shapefile format limits attribute data to 255 characters:

https://github.com/archesproject/arches/blob/b0e54e01251e54adda7c3b10a31bb8670bdfacb0/arches/app/utils/data_management/resources/formats/shpfile.py#L142-L148

This seems like a data accessibility issue, since users are not currently able to see >255 characters of their data in shapefile exports. This limit should be configurable via settings.py.

**Comment by bferguso** [comment]
I thought the Shapefile format, by definition, was limited to 255 characters for attribute data?

**Comment by adamlodge** [comment]
I can echo Brett's question as an assertion. The problem is the limitation
of the shapefile format itself - not the Arches export functionality. To
meet the need you describe, you'd have to pick a different format to export
to.

Shapefile - Wikipedia https://share.google/njW9yxJSC3GwiYQIz

**Comment by samuelscandrett** [comment]
Thanks both, I also just realised that shapefiles have a hard limit of 255 characters after reading about it [here](https://desktop.arcgis.com/en/arcmap/latest/manage-data/shapefiles/geoprocessing-considerations-for-shapefile-output.htm#GUID-A10ADA3B-0988-4AB1-9EBA-AD704F77B4A2) .

Closing this issue!
